Nodejs/Product/ProjectWizard/ImportWizard.cs

```diff
@@ -15,74 +14,75 @@ namespace Microsoft.NodejsTools.ProjectWizard
 {
     public sealed class NewProjectFromExistingWizard : IWizard
     {
-        public static Boolean IsAddNewProjectCmd { get; set; }
+        public static bool IsAddNewProjectCmd { get; set; }
         public void BeforeOpeningFile(EnvDTE.ProjectItem projectItem) { }
         public void ProjectFinishedGenerating(EnvDTE.Project project) { }
         public void ProjectItemFinishedGenerating(EnvDTE.ProjectItem projectItem) { }
         public void RunFinished() { }
 
         public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams)
         {
-            var dte = automationObject as DTE;
-            if (dte == null)
+            var provider = automationObject as Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
+            if (provider == null)
             {
-                var provider = automationObject as Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
-                if (provider != null)
-                {
-                    dte = new ServiceProvider(provider).GetService(typeof(DTE)) as DTE;
-                }
+                MessageBox.Show(ProjectWizardResources.ErrorNoDte, SR.ProductName);
+                throw new WizardBackoutException();
             }
 
-            bool addingNewProject = false;
-            if (dte == null)
+            using (var serviceProvider = new ServiceProvider(provider))
             {
-                MessageBox.Show(ProjectWizardResources.ImportWizzardCouldNotStartNotAutomationObjectErrorMessage, SR.ProductName);
-            }
-            else
-            {
-                // https://nodejstools.codeplex.com/workitem/462
-                // we need to make sure our package is loaded before invoking our command
-                Guid packageGuid = new Guid(Guids.NodejsPackageString);
-                IVsPackage package;
-                ((IVsShell)Package.GetGlobalService(typeof(SVsShell))).LoadPackage(
-                    ref packageGuid,
-                    out package
-                );
+                var hr = EnsurePackageLoaded(serviceProvider);
+                if (ErrorHandler.Failed(hr))
+                {
+                    MessageBox.Show(string.Format(ProjectWizardResources.ImportWizardCouldNotStartFailedToLoadPackage, hr), "Visual Studio");
+                    throw new WizardBackoutException();
+                }
+                var uiShell = (IVsUIShell)serviceProvider.GetService(typeof(SVsUIShell));
 
-                System.Threading.Tasks.Task.Factory.StartNew(() =>
+                var projName = replacementsDictionary["$projectname$"];
+                replacementsDictionary.TryGetValue("$specifiedsolutionname$", out var solnName);
```

Comment:
I don't think you can assume that this sets null on failure.  I'd check the return value in the `if` below.